### DISCUSSION TITLE
How can i use Auth0 with vercel serverless

### DISCUSSION THREAD

**Discussion by wangcode** [vercel/vercel#6754] [Help] [answered]
front is SPA with react+auth0-react
backend is nodejs 
how to auth user and get user info from auth0

**Comment by leerob** [comment, marked as answer]
Hey! I would recommend starting from this example, which handles React on the frontend and Node.js on the backend (through Next.js API Routes).

https://github.com/vercel/next.js/tree/canary/examples/auth0

**Reply by wangcode** [reply]
thanks for your answer
you mean i need use the nextjs-auth0 package even i don't use the nextjs?

**Reply by leerob** [reply]
You don't need to use Next.js, but it will be an easier experience since you can handle both your frontend and backend in the same project.

Of course, you can still use the normal React SDK too: https://auth0.com/docs/libraries/auth0-react

Example: https://github.com/auth0/auth0-react/tree/master/examples/cra-react-router

**Reply by wangcode** [reply]
look like i need to use nextjs-auth0 for my backend.
it's there have any other way (or example) to get user profile on the backend in vercel serverless?
thank you again.

**Reply by leerob** [reply]
If you want to use serverless, then yes I would recommend using Next.js for the backend with their SDK. Otherwise, you can deploy a Node.js server to any serverful hosting provider (e.g. Railway, Heroku, AWS).

**Reply by wangcode** [reply]
got it!
Thank you for your patience, have a nice day : )
